planner/core/exhaust_physical_plans.go

```diff
@@ -1595,6 +1595,27 @@ func (p *LogicalJoin) tryToGetIndexJoin(prop *property.PhysicalProperty) (indexJ
 	return append(allLeftOuterJoins, allRightOuterJoins...), false
 }
 
+func checkChildFitBC(p Plan) bool {
+	if p.statsInfo().HistColl == nil {
+		return p.statsInfo().Count() < p.SCtx().GetSessionVars().BroadcastJoinThresholdCount
+	}
+	avg := p.statsInfo().HistColl.GetAvgRowSize(p.SCtx(), p.Schema().Columns, false, false)
+	sz := avg * float64(p.statsInfo().Count())
+	return sz < float64(p.SCtx().GetSessionVars().BroadcastJoinThresholdSize)
+}
+
+func (p *LogicalJoin) shouldUseMPPBCJ() bool {
+	if p.ctx.GetSessionVars().BroadcastJoinThresholdSize == 0 || p.ctx.GetSessionVars().BroadcastJoinThresholdCount == 0 {
+		return p.ctx.GetSessionVars().AllowBCJ
+	}
+	if p.JoinType == LeftOuterJoin || p.JoinType == SemiJoin || p.JoinType == AntiSemiJoin {
+		return checkChildFitBC(p.children[1])
+	} else if p.JoinType == RightOuterJoin {
+		return checkChildFitBC(p.children[0])
+	}
+	return checkChildFitBC(p.children[0]) || checkChildFitBC(p.children[1])
```

Comment:
We also check the two sides if JoinType is `LeftOuterSemiJoin` and `AntiLeftOuterSemiJoin`?

Reply:
No need, it the child is outer semi join, I think they can not be the mpp task.

Reply:
No need , doesn't it make sense ?